### PR TITLE
Add prefix functionality to scopes to facilitate multiple state machines on one model.

### PR DESCRIPTION
**Pull request by nicholalexander**
When using multiple state machines on a single model, one might want more control over their scopes.  In order to allow that, this PR adds handling a hash with some prefix options that works similar to how delegate in rails works.  You can pass in `scopes: { prefix: true }` to get your scope prefixed with your attribute or `scopes: { prefix: 'some_custom_prefix' }` to get your scope prefixed with a custom prefix.

This PR also adds associated tests and updates the Readme.

### CODE REVIEW

File: README.md
Comment (smudge) [inline README.md:224]:
```suggestion
steady_state :temperature, scopes: { prefix: true } do
```

I might propose this style (similar to Rails' `delegate` prefix option), where `prefix: true` would default to the name of the state field, and `prefix: 'something_else'` would allow you to override it with something else.

Comment (nicholalexander) [reply README.md:224]: Great!  And also, I really like the idea to make it like rails delegate..  When Ray and I were talking about this, he also thought about `prefix: true` but we didn't make the connection to do it all three ways.  I will update the PR and write a description shortly!

File: lib/steady_state/attribute.rb
Comment (nicholalexander) [inline lib/steady_state/attribute.rb:63]: Sort of dense!  We are handling the following situations though:

```
scopes: true
scopes: false
scopes: { prefix: true }
scopes: { prefix: 'some_custom_prefix' }
```

Interestingly, `scopes: { prefix: false }` behaves like `scopes: true`.  Which I think is what I would expect.  

I also experimented a little with having a default `else` branch that raised if the key wasn't present.  For example, if you recently returned home from brunch and did `scopes: { prixefixe: true }` it just silently defaults to behavior equivalent to `scopes: true`.  If we wanted to be more supportive of users, a little defensive error there might be helpful such that we'd get:

```suggestion
          prefix = if scopes == true
                     ''
                   elsif scopes.is_a?(Hash) && scopes[:prefix]
                     "#{scopes[:prefix] == true ? attr_name : scopes[:prefix]}_"
                   else
                      raise ArgumentError, 'Invalid scopes option. Must be a Hash with a :prefix key or a boolean'
                   end
```

Comment (smudge) [reply lib/steady_state/attribute.rb:63]: On newer Rubies I tend to take nested hash options (like `:prefix`) and feed them directly into a new method as keyword arguments. This allows me to specify required/optional kwargs and get Ruby's built-in `ArgumentError` if an option is typoed.

E.g.:

```ruby
def build_prefix(prefix: false)
  prefix ? "#{prefix == true ? attr_name : prefix}_" : ""
end
```

And then the example above becomes:
```suggestion
          scopes = scopes.is_a?(Hash) ? scopes : {}
          prefix = build_prefix(**scopes)
```

**Review by smudge — DISMISSED** [review]
domain LGTM

**Review by smudge — CHANGES_REQUESTED** [review]
platform TAFN -- just the one note! Mostly a thought about how to get an `ArgumentError` idiomatically, but overall, this looks great.

Before you merge, you may also want to bump the version number so that I can cut you a release right away. (Not that you need this urgently or anything... apologies for the slow review. 😅)

File: lib/steady_state/attribute.rb
Comment (nicholalexander) [reply lib/steady_state/attribute.rb:63]: Oh, nice.  Will update!

**Comment by nicholalexander** [comment]
> platform TAFN -- just the one note! Mostly a thought about how to get an `ArgumentError` idiomatically, but overall, this looks great.
> 
> Before you merge, you may also want to bump the version number so that I can cut you a release right away. (Not that you need this urgently or anything... apologies for the slow review. 😅)

Will also bump version number!  No apologies necessary!

**Review by smudge — APPROVED** [review]
domain LGTM && platform LGTM!

Looks like a dependabot merge may have messed up the build matrix gemfiles. You can run `bundle exec appraisal install` to regenerate those, or we can do that in a separate PR.

**Comment by smudge** [comment]
@nicholalexander fixed the build issue over here:
https://github.com/Betterment/steady_state/pull/17

File: lib/steady_state/version.rb
Comment (smudge) [inline lib/steady_state/version.rb:4]: I think you'll need to re-run `bundle install` and `bundle exec appraisal install` (and `bundle exec rubocop -A` if appraisal introduces linter errors, lol), and then commit those changes. That should be the last piece of this that will unblock your build.

Comment (nicholalexander) [reply lib/steady_state/version.rb:4]: Thanks!  I started on this yesterday and got sidetracked - should be incoming!

**Review by smudge — APPROVED** [review]
domain LGTM && platform LGTM!

**Pull request merged**
